fairscale/optim/layerwise_gradient_scaler.py

```diff
@@ -0,0 +1,278 @@
+import logging
+from typing import List, Tuple
+
+import torch
+import torch.nn as nn
+
+
+class LayerInfo:
+    """
+    A class to record the layer attributes.
+    """
+
+    def __init__(self, name: str, layer: nn.Module, scale: float = 1.0, scale_layer: bool = False) -> None:
+        """
+        layer_name: name of the layer e.g. fc1, conv1, relu1
+        layer_type: type of the layer e.g. Linear, Conv2d, ReLU
+        scaling_factor: user configurable scaling factor for the layer, defaults to 1.0
+        found_inf_or_nan: a boolean indicating if any parameter of layers' gradient contains inf/nan
+        growth_tracker: tracks number of step since last time scale was increased
+        scale_layer: a boolean indicating if the layer should be scaled or not
+        """
+        self.layer_name = name
+        self.layer_type = layer
```

Comment:
should this be `layer` instead of `layer_type`? type makes me thing we are storing a string rather than the layer itself.

Reply:
layer_name is the name as a string. layer_type is Linear, Conv2d etc. I gave examples on line 16.

Reply:
I guess we assign to layer which is a nn.Module hence my question.

Reply:
I can change it if that improves readability. Do let me know.

At this point I'm deep in the trenches with this code, so a fresh perspective on naming convention (and anything else) is very welcome :)

Reply:
From my perspective its a layer your are tracking and not its type. No need to change it since it is a nit.

Reply:
changed